src/Silex/Provider/TranslationServiceProvider.php

```diff
@@ -53,6 +58,8 @@ public function register(Application $app)
             return new MessageSelector();
         });
 
+        $app['translator.cache_dir'] = null;
+        $app['translator.debug'] = $app['debug'];
```

Comment:
this should use a lazy definition to allow it to take into account changes to the global debug mode done after the registration of the provider